src/common.rs

```diff
@@ -64,6 +63,24 @@ pub fn set_fd_close_exec(fd: RawFd) -> Result<RawFd> {
     Ok(fd)
 }
 
+// SOCK_CLOEXEC flag is Linux specific
+#[cfg(target_os = "linux")]
+pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::SOCK_CLOEXEC;
+#[cfg(not(target_os = "linux"))]
+pub(crate) const SOCK_CLOEXEC: SockFlag = SockFlag::empty();
+
+#[cfg(target_os = "linux")]
+fn make_addr(host: &str) -> Result<UnixAddr> {
+    UnixAddr::new_abstract(host.as_bytes()).map_err(err_to_others_err!(e, ""))
+}
+
+#[cfg(not(target_os = "linux"))]
+fn make_addr(host: &str) -> Result<UnixAddr> {
+    UnixAddr::new(host).map_err(err_to_others_err!(e, ""))
+}
+
+#[cfg_attr(target_os = "linux", allow(unreachable_patterns))]
+#[cfg_attr(not(target_os = "linux"), allow(unused_variables))]
```

Comment:
I think these two allow attributes are dangerous because disable these checks may blind the lint when new error was introduced.

How about we do the trick in parse_host, detect macos and return error when using vsock

Reply:
I've updated the PR to get rid of these suppressions.

I's not quite possible to return error on vsock as its symbols are excluded from nix crate for non linux (see my comment below).

Reply:
ok